Assets/Scripts/EnemyController.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+public class EnemyController : MonoBehaviour
+{
+    public Transform[] patrolPoints;
+    public float moveSpeed;
+    public int patrolDestination;
+    private bool isAlive = true;
+    private Animator animator;
+    private bool isInRangeAnimationPlaying = false;
+    private bool isDeathAnimationPlaying = false;
+
+    private void Start()
+    {
+        animator = GetComponent<Animator>();
+    }
+
+    private void Update()
+    {
+        if (isAlive)
+        {
+            EnemyMovement();
+        }
+    }
+
+    private void EnemyMovement()
+    {
+        Vector3 scale = transform.localScale;
```

Comment:
No need to create a Vector3 scale every frame.
You can create a separate method for setting the scale and call that in the if and else condition.